apps/web/src/stores/commits.ts

```diff
@@ -0,0 +1,39 @@
+import { useCallback } from 'react'
+
+import type { Commit } from '@latitude-data/core'
+import { createCommitAction } from '$/actions/commits/create'
+import { getCommitsAction } from '$/actions/commits/fetch'
+import useSWR, { SWRConfiguration } from 'swr'
+import { useServerAction } from 'zsa-react'
+
+export default function useCommits(
+  { projectId }: { projectId: number },
+  opts?: SWRConfiguration,
+) {
+  const key = '/api/commits'
+  const { execute } = useServerAction(createCommitAction)
+  const {
+    data = [],
+    mutate,
+    ...rest
+  } = useSWR<Commit[]>(
+    key,
+    async () => {
+      const [commits] = await getCommitsAction({ projectId })
```

Comment:
We should implement a convention on how to deal with errors on actions in the frontend. We have errors in the returned value here on outside. Let's pick one and stick to it. Specially for actions that are "GET" like this